### PR TITLE
Add docs to conftest.py

### PR DESCRIPTION
**Pull request by Seanny123**
Addresses some of confusion I had previous in #1071

### CODE REVIEW

**Comment by jgosmann** [comment]
LGTM :cake:

File: nengo/conftest.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def load_class(fully_qualified_name):
 
 @pytest.fixture(scope="session")
 def Simulator(request):
-    """the Simulator class being tested.
+    """The Simulator class being tested.
```
Comment (tbekolay) [inline nengo/conftest.py]: FYI this used to be the convention used by py.test for fixture docstrings (which are printed when you do `py.test --fixtures`) but it seems as though they've switched to sentence case :+1:

**Comment by tbekolay** [comment]
Pushed a fixup commit to be closer to [PEP-257 style](https://www.python.org/dev/peps/pep-0257/). I found [pydocstyle](https://github.com/PyCQA/pydocstyle) which I'll get set up so that this is checked in CI. Take a look @Seanny123 and if it looks good to you I'll merge.

**Comment by Seanny123** [comment]
LGTM. I didn't know there was a style for docs! I'm glad it will be checked in the CI from now on.

**Pull request merged**